modules/parent-join/src/main/java/org/elasticsearch/join/aggregations/InternalChildren.java

```diff
@@ -11,15 +11,15 @@
 
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.search.aggregations.InternalAggregations;
-import org.elasticsearch.search.aggregations.bucket.InternalSingleBucketAggregation;
+import org.elasticsearch.search.aggregations.bucket.SingleBucketAggregation;
 
 import java.io.IOException;
 import java.util.Map;
 
 /**
  * Results of the {@link ParentToChildrenAggregator}.
  */
-public class InternalChildren extends InternalSingleBucketAggregation {
```

Comment:
The motivation in cleaning this up shoud become visible here, all but one implementation of `SingleBucketAggregation` are noop overrides. We can (with a bit of BwC code for the transport layer) fold all of these into a single class and save both a lot of code but also make serialization a lot more compact (and faster!).